pkg/mimir/modules.go

```diff
@@ -462,6 +463,12 @@ func (t *Mimir) initStoreQueryables() (services.Service, error) {
 	}
 }
 
+func (t *Mimir) initActiveGroupsCleanupService() (services.Service, error) {
+	t.ActiveGroupsCleanup = util.NewActiveGroupsCleanupWithDefaultValues(t.Cfg.MaxGroupsPerUser, t.Distributor.RemoveGroupMetricsForUser, t.Ingester.RemoveGroupMetricsForUser)
```

Comment:
:exclamation: This will not work and will panic after the cleanup interval. You're storing references to nil Distributor and Ingester, as they're not initialized yet when this service is started.
See this example:

https://go.dev/play/p/FpK0JeU7ZJZ

Just create an empty service here, add a method to the service:
```
type UserGroupMetricsCleaner interface {
	RemoveGroupMetricsForUser(user, group string)
}

func (ag *ActiveGroupsCleanupService) Register(metricsCleaner UserGroupMetricsCleaner) {
	ag.mtx.Lock()
	defer ag.mtx.Unlock()
	ag.cleaners = append(ag.cleaners, metricsCleaner)
}
```

And do `service.Register(t.Distributor)` once the distributor is initialized.